modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2476,35 +2476,44 @@ def drop(self, index=None, columns=None, errors: str = "raise"):
 
     def duplicated(self, **kwargs):
         def _compute_hash(df):
-            return df.apply(
+            result = df.apply(
                 lambda s: hashlib.new("md5", str(tuple(s)).encode()).hexdigest(), axis=1
-            ).to_frame()
+            )
+            if isinstance(result, pandas.Series):
+                result = result.to_frame(
+                    result.name
+                    if result.name is not None
+                    else MODIN_UNNAMED_SERIES_LABEL
+                )
+            return result
 
         def _compute_duplicated(df):
-            return df.duplicated(**kwargs).to_frame()
+            result = df.duplicated(**kwargs)
+            if isinstance(result, pandas.Series):
+                result = result.to_frame(
+                    result.name
+                    if result.name is not None
+                    else MODIN_UNNAMED_SERIES_LABEL
+                )
+            return result
 
-        new_index = self._modin_frame._index_cache
-        new_columns = [MODIN_UNNAMED_SERIES_LABEL]
-        if len(self.columns) > 1:
-            # if the number of columns we are checking for duplicates is larger than 1,
-            # we must hash them to generate a single value that can be compared across rows.
-            hashed_modin_frame = self._modin_frame.apply_full_axis(
-                1,
-                _compute_hash,
-                new_index=new_index,
-                new_columns=new_columns,
-                keep_partitioning=False,
+        if len(self._modin_frame.column_widths) > 1:
```

Comment:
`column_widths` may not be available at this point, meaning that this line would trigger its computation, can we use `._partitions.shape` here? it seems to have the same result
```suggestion
        if self._modin_frame._partitions.shape[1] > 1:
```

Reply:
self._modin_frame._partitions.shape[1] may have some non-empty partitions and some other empty partitions so we may get unnecessary computation here. I propose to leave as is.

Reply:
`len(self._modin_frame.column_widths)` always equals to the number of splits across the row axis, meaning that it always equals to `._partitions.shape[1]` isn't it?

Reply:
I mean, if there are empty partitions in the frame, they would still be visible in column widths: `[4, 0, 0, 0]`, what's the point of using them?